lib/debootstrap.sh

```diff
@@ -686,12 +692,17 @@ PREPARE_IMAGE_SIZE
 		echo "$rootfs / ${mkfs[$ROOTFS_TYPE]} defaults,noatime${mountopts[$ROOTFS_TYPE]} 0 1" >> $SDCARD/etc/fstab
 	fi
 	if [[ -n $bootpart ]]; then
-		display_alert "Creating /boot" "$bootfs on ${LOOP}p${bootpart}"
-		check_loop_device "${LOOP}p${bootpart}"
-		mkfs.${mkfs[$bootfs]} ${mkopts[$bootfs]} ${mkopts_label[$bootfs]:+${mkopts_label[$bootfs]}"$BOOT_FS_LABEL"} ${LOOP}p${bootpart} >> "${DEST}"/${LOG_SUBPATH}/install.log 2>&1
-		mkdir -p $MOUNT/boot/
-		mount ${LOOP}p${bootpart} $MOUNT/boot/
-		echo "UUID=$(blkid -s UUID -o value ${LOOP}p${bootpart}) /boot ${mkfs[$bootfs]} defaults${mountopts[$bootfs]} 0 2" >> $SDCARD/etc/fstab
+		local bootdevice="${LOOP}p${bootpart}"
+		display_alert "Creating boot fs" "${bootfs} on ${bootdevice}"
+		check_loop_device ${bootdevice}
+		mkfs.${mkfs[$bootfs]} ${mkopts[$bootfs]:=} ${mkopts_label[$bootfs]:+${mkopts_label[$bootfs]}${BOOT_FS_LABEL}} ${bootdevice} >> ${DEST}/${LOG_SUBPATH}/install.log 2>&1
+		mkdir -p ${MOUNT}/boot
+		[[ ${bootfs} == fat ]] && {
+			local bfsmntopt="${mkfs[$bootfs]:+ -t vfat}"' '
+			display_alert "Mount command line parameter with the type of the boot file system:" "${bfsmntopt}"
+		}
+		mount ${bfsmntopt:=} ${bootdevice} ${MOUNT}/boot
```

Comment:
Necessity? mount will detect the type of filesystem.

Reply:
Agree. Got there because of a problem that later proved to be caused by something else  (the boot partition not getting filled). But does not hurt either as the script just knows what the fs type is, so mount may find the proper fs type quicker that way.

Reply:
Keep it simple. It's really strange to add a special handle only for fat32.